slack_global_report.py

```diff
@@ -0,0 +1,241 @@
+        text += "The top ten posters contributed *{:.1f}%* of all messages (lower is better)\n".format(stats['topten messages'])
+        text += "The top *{}* posters (higher is better) accounted for about 50% of total volume\n".format(stats['50percent of words'])
+        blocks.append(self.sf.text_block(text))
+        w = stats['words']
+        m = stats['messages']
+        # it = interim text
+        it = "People posted *{:,}* words in *{:,}* messages (approximately *{:,}* pages), "
+        it += "or about *{:.1f}* words per message, *{:.1f}* words per poster, "
+        it += "or *{:.1f}* messages per poster"
+        text = it.format(int(w), int(m), int(w / 500), w/m, w/posters, m/posters)
+        blocks.append(self.sf.text_block(text))
+        blocks.append(self.sf.divider())
+        return blocks
+
+    def top_channels(self, ur, pur):
+        blocks = []
+        top = 20
+        header = "*Top {} Channels*".format(top)
+        blocks.append(self.sf.text_block(header))
+        channels = ur['channels']
+        cids = list(channels.keys())[:top]
+        cinfo = ur['channel_info']
+        cstats = ur['channel_stats']
+        cusers = ur['channel_user']
+        for idx, channel in enumerate(cids):
+            it = "{}. {} ".format(idx + 1, self.sf.show_cid(channel))
+            ci = cinfo[channel]
+            cs = cstats[channel]
+            cu = cusers[channel]
+            if ci['new']:
+                it += " (new)"
+            it += "{:,} members ".format(ci['members'])
+
+            m = channels[channel][0]
+            w = channels[channel][1]
+            p = len(cu)
+            it += "*{:,}* posters *{:,}* words *{:,}* messages ".format(p, w, m)
+            it += "*{:.1f}* words/poster ".format(w/p)
+            it += "*{:.1f}%* of total traffic, ".format(cs['percent'])
+            it += "*{:.1f}%* cumulative of total ".format(cs['cpercent'])
+            blocks.append(self.sf.text_block(it))
+        blocks.append(self.sf.divider())
+        return blocks
+
+    def top_users(self, ur, pur):
+        blocks = []
+        top = 20
+        header = "*Top {} Users*".format(top)
+        blocks.append(self.sf.text_block(header))
+        stats = ur['statistics']
+        us = ur['user_stats']
+        uids = stats['50percent users for words'][:top]
+        for idx, uid in enumerate(uids):
+            usu = us[uid]
+            m = usu['count'][0]
+            w = usu['count'][1]
+            per = usu['percent_of_words']
+            cper = usu['cum_percent_of_words']
+            rphw = usu['reactions'] * 100.0 / w
+            w_per_m = w / m
+            t = usu['thread_messages']
+            it = "{}. *{}* ".format(idx + 1, self.sf.show_uid(uid))
+            it += "*{:,}* words *{:,}* messages *{:.1f}* w/m ".format(w, m, w_per_m)
+            it += "*{:.1f}* rphw ".format(rphw)
+            it += "*{:.1f}%* *{:.1f}%* cumulative\n".format(per, cper)
```

Comment:
Is this cumulative of total? If so, can you change it to be cumulative of total?

Reply:
Done